internal/resolution/resolver.go

```diff
@@ -114,11 +115,40 @@ func (r Resolver) searchDirs(fileSet map[string]bool, dirs []string, exclusions
 			return err
 		}
 		for _, fileGroup := range fileGroups.ToSlice() {
-			if fileGroup.HasFile() && !fileGroup.HasLockFiles() {
+			shouldGenerate, err := shouldGenerateLock(fileGroup, regenerate)
+			if err != nil {
+				continue
+			}
```

Comment:
Right now we completely ignore any errors here (probably only invalid flag), should I remove that error then or do we want to do something with a possible error here?